src/index.ts

```diff
@@ -19,8 +19,9 @@ async function run() {
     await workingDirectory.mkdir();
     const optionsFile = getInput(Options.OptionsFile);
     const cacheMayBeUsed = optionsFile || shouldUseCache();
+    const workspace = getEnvOrThrow('GITHUB_WORKSPACE');
     if (cacheMayBeUsed) {
-        await checkRepositoryIsNotShallow();
+        await checkRepositoryIsNotShallow(workspace);
```

Comment:
This and related refactors that follow in the same file are not necessary to implement this PR, but I like them. Good work! :heart: